lite/dbprovider.go

```diff
@@ -0,0 +1,260 @@
+package lite
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	amino "github.com/tendermint/go-amino"
+	cryptoAmino "github.com/tendermint/tendermint/crypto/encoding/amino"
+	dbm "github.com/tendermint/tendermint/libs/db"
+	log "github.com/tendermint/tendermint/libs/log"
+	lerr "github.com/tendermint/tendermint/lite/errors"
+	"github.com/tendermint/tendermint/types"
+)
+
+type DBProvider struct {
+	logger log.Logger
+	label  string
+	db     dbm.DB
+	cdc    *amino.Codec
+	limit  int
+}
+
+func NewDBProvider(label string, db dbm.DB) *DBProvider {
+	//db = dbm.NewDebugDB("db provider "+cmn.RandStr(4), db)
+	cdc := amino.NewCodec()
+	cryptoAmino.RegisterAmino(cdc)
+	dbp := &DBProvider{
+		logger: log.NewNopLogger(),
+		label:  label,
+		db:     db,
+		cdc:    cdc,
+	}
+	return dbp
+}
+
+func (dbp *DBProvider) SetLogger(logger log.Logger) {
+	dbp.logger = logger.With("label", dbp.label)
+}
+
+func (dbp *DBProvider) SetLimit(limit int) *DBProvider {
+	dbp.limit = limit
+	return dbp
+}
+
+// Implements PersistentProvider.
+func (dbp *DBProvider) SaveFullCommit(fc FullCommit) error {
```

Comment:
It might be useful having a short design document about DBProvider.

Reply:
It's not optimal but it works.  I'm more concerned about getting something that works, and we can optimize the data model later.

Reply:
Fair. We are trying (as much as possible) to respect development process where non-trivial code changes are preceded with a design document (ADR). ADR can be super simple, but it helps us being explicit about current design and possible future improvements. It should help especially for people that read code after.

Reply:
Sorry for breaking protocol.  I agree that the next set of changes here should start with an ADR.  In the interest of time I'll merge this first and create an issue for it.

Reply:
https://github.com/tendermint/tendermint/issues/2133